docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -1,763 +1,162 @@
+  <ContentSpotlight src="/static/images/eas-build/tests/02-explore.png" className="max-w-[360px]" />
+</div>
 
-- For Android, you have to manually start the emulator before running the tests as `detox` occasionally encounters issues initiating the emulator independently, which can cause a hang during the first test of your suite. After completing the `detox test`, there is a command that kills the previously started emulator.
-- For iOS, the only command is `detox test`.
+Let's create two simple Maestro flows for the example app. Start by creating a directory called **maestro** in the root of your project directory. This directory will contain the flows that you will configure and should be at the same level as **eas.json**.
 
-```sh eas-hooks/eas-build-on-success.sh
-#!/usr/bin/env bash
 
-function cleanup()
-{
-  echo 'Cleaning up...'
-  if [[ "$EAS_BUILD_PLATFORM" == "android" ]]; then
-    # Kill emulator
-    adb emu kill &
-  fi
-}
+Inside it, create a new flow called **home.yaml**. This flow will launch the app and assert that the text "Welcome!" is visible on the home screen.
 
-if [[ "$EAS_BUILD_PROFILE" != "test" ]]; then
-  exit
-fi
-
-# Fail if anything errors
-set -eox pipefail
-# If this script exits, trap it first and clean up the emulator
-trap cleanup EXIT
-
-ANDROID_EMULATOR=pixel_4
-
-if [[ "$EAS_BUILD_PLATFORM" == "android" ]]; then
-  # Start emulator
-  $ANDROID_SDK_ROOT/emulator/emulator @$ANDROID_EMULATOR -no-audio -no-boot-anim -no-window -use-system-libs 2>&1 >/dev/null &
-
-  # Wait for emulator
-  max_retry=10
-  counter=0
-  until adb shell getprop sys.boot_completed; do
-    sleep 10
-    [[ counter -eq $max_retry ]] && echo "Failed to start the emulator!" && exit 1
-    counter=$((counter + 1))
-  done
-
-  # Execute Android tests
-  if [[ "$EAS_BUILD_PROFILE" == "test" ]]; then
-    detox test --configuration android.release
-  fi
-else
-  # Execute iOS tests
-  if [[  "$EAS_BUILD_PROFILE" == "test" ]]; then
-    detox test --configuration ios.release
-  fi
-fi
+```yaml maestro/home.yaml
+appId: dev.expo.eastestsexample # This is an example app id. Replace it with your app id.
+---
+- launchApp
+- assertVisible: 'Welcome!'
 ```
 
-Update **package.json** to use [EAS Build hooks](/build-reference/npm-hooks/) to run the above scripts on EAS Build:
+Next, create a new flow called **expand_test.yaml**. This flow will open the "Explore" screen in the example app, click on the "File-based routing" collapsible, and assert that the text "This app has two screens" is visible on the screen.
 
-```json package.json
-{
-  "scripts": {
-    "eas-build-pre-install": "./eas-hooks/eas-build-pre-install.sh",
-    "eas-build-on-success": "./eas-hooks/eas-build-on-success.sh"
-  }
-}
+```yaml maestro/expand_test.yaml
+appId: dev.expo.eastestsexample # This is an example app id. Replace it with your app id.
+---
+- launchApp
+- tapOn: 'Explore.*'
+- tapOn: '.*File-based routing'
+- assertVisible: 'This app has two screens.*'
```

Comment:
Should we also let people know how to test the workflows they're developing on their machines?

Reply:
Yeah, good idea! Do you mean by running `maestro test meastro/home.yml`?